sqlmesh/core/analytics/collector.py

```diff
@@ -0,0 +1,351 @@
+        additional_args = {}
+        if cicd_bot_config is not None and getattr(cicd_bot_config, "FIELDS_FOR_ANALYTICS", None):
+            additional_args["cicd_bot_config"] = cicd_bot_config.dict(
+                include=cicd_bot_config.FIELDS_FOR_ANALYTICS, mode="json"
+            )
+        self._on_command(
+            "CICD_COMMAND",
+            command_name,
+            command_args,
+            parent_command_names=parent_command_names,
+            **additional_args,
+        )
+
+    def on_cli_command(
+        self,
+        *,
+        command_name: str,
+        command_args: t.Collection[str],
+        parent_command_names: t.Collection[str],
+    ) -> None:
+        """Called when a CLI command is executed.
+
+        Args:
+            command_name: The name of the command.
+            command_args: The arguments of the command.
+            parent_command_names: The names of the parent commands.
+        """
+        self._on_command(
+            "CLI_COMMAND", command_name, command_args, parent_command_names=parent_command_names
+        )
+
+    def on_magic_command(self, *, command_name: str, command_args: t.Collection[str]) -> None:
+        """Called when a Notebook magic command is executed.
+
+        Args:
+            command_name: The name of the command.
+            command_args: The arguments of the command.
+        """
+        self._on_command("MAGIC_COMMAND", command_name, command_args)
+
+    def on_python_api_command(self, *, command_name: str, command_args: t.Collection[str]) -> None:
+        """Called when a Python method is called directly.
+
+        Args:
+            command_name: The name of the command.
+            command_args: The arguments of the command.
+        """
+        self._on_command("PYTHON_API_COMMAND", command_name, command_args)
+
+    def on_project_loaded(
+        self,
+        *,
+        project_type: str,
+        models_count: int,
+        audits_count: int,
+        standalone_audits_count: int,
+        macros_count: int,
+        jinja_macros_count: int,
+        load_time_sec: float,
+        state_sync_fingerprint: str,
+        project_name: str,
+    ) -> None:
+        """Called when a project is loaded.
+
+        Args:
+            project_type: The type of the project. Eg. "dbt" or "native".
+            models_count: The number of models in the project.
+            audits_count: The number of audits in the project.
+            standalone_audits_count: The number of standalone audits in the project.
+            macros_count: The number of macros in the project.
+            jinja_macros_count: The number of Jinja macros in the project.
+            load_time_sec: The time it took to load the project in (fractional) seconds.
+            state_sync_fingerprint: The fingerprint of the state sync configuration.
+            project_name: The name of the project.
+        """
+        dbt_version = None
+        project_type = project_type.upper()
```

Comment:
just because i'm ocd, let's use lower(), since dbt and native should just be lower case, also we use lower for engine types